src/server/route.rs

```diff
@@ -62,6 +64,19 @@ impl<'a, State: 'static> Route<'a, State> {
         self
     }
 
+    /// Nest a [`Server`] at the current path.
+    ///
+    /// [`Server`]: struct.Server.html
+    pub fn nest_service(&mut self, service: crate::Server<State>) -> &mut Self
```

Comment:
I wonder if we could shorten this to `nest`. I'm still wondering if we could implement `Service` on `Server` directly, at which point the word `service` will likely disappear from most places. `nest` still feels quite clear to me.

Reply:
Well, I think it's okay to leave this as is, because `tide::Server` is more like a builder type for `tide::Service`. This seems natural for me, how do you think?

And yeah, `nest` seems clear enough; maybe we can remove existing method and rename this to `nest`?

Reply:
Renaming to `nest` sounds good.

Also I've been wondering if we need a distinction between `Server` and `Service`. It seems that if they could be the same struct that would allow us to consolidate the API a bit. But that's something for the future.